### PR TITLE
[BugFix](Jdbc Catalog) Fix null pointer exception in JdbcExecutor

### PR DESCRIPTION
**Pull request by BePPPower**
# Proposed changes

Issue Number: close #xxx

This pr do two things:
1. fix: 
    It use `column[0]` to judge class type in JdbcExecutor, but column[0] may be null !

3. enhencement:
     In the original logic, all fields in jdbc catalog table will be set Nullable. Howerver, it is inefficient for nullable fields. Actually, we can know if the fields in data source table is nullable throught jdbc. So we can set the corresponding fields in doris jdbc catalog to nullable or not.

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by morningman** [comment]
run buildall

File: fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java [hunk 1]
```diff
@@ -315,11 +316,19 @@ private void shortPutToByte(Object[] column, boolean isNullable, int numRows, lo
     public void copyBatchTinyIntResult(Object columnObj, boolean isNullable, int numRows, long nullMapAddr,
             long columnAddr) {
         Object[] column = (Object[]) columnObj;
-        if (column[0] instanceof BigDecimal) {
+        int firstNotNullIndex = 0;
+        if (isNullable) {
+            firstNotNullIndex = getFirstNotNullObject(column, numRows, nullMapAddr);
+        }
```
Comment (zhangstar333) [inline fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java:322]: here, if the column all data is null, so the firstNotNullIndex==numRows, and then check column[firstNotNullIndex], will be out of range?

Comment (BePPPower) [reply fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java:322]: You are right!

Comment (BePPPower) [reply fe/java-udf/src/main/java/org/apache/doris/udf/JdbcExecutor.java:322]: done

**Comment by BePPPower** [comment]
run buildall

**Comment by BePPPower** [comment]
run p0

**Review by zhangstar333 — DISMISSED** [review]

**Review by zhangstar333 — APPROVED** [review]

**Comment by BePPPower** [comment]
run buildall

**Review by morningman — APPROVED** [review]
LGTM

**Pull request merged**